analyses/copy_number_consensus_call/Snakefile

```diff
@@ -193,6 +193,35 @@ rule combine_merge_paired_cnv:
         "| bedtools merge -c 4,5,6,7,8,9 -o collapse,collapse,collapse,distinct,distinct,collapse "
         "> {output.merged}"
 
+rule remove_inset_calls:
+    ## Sometimes calls are inset within one another: one del call within the range of a dup call, or vice versa.
+    ## To fix this, we will subtract del calls from dup calls, which splits them and removes the overlap.
+    ## We then need to add back any dup calls that may have been fully enclosed in del calls, which we do
+    ## by intesecting dup and del calls, then intersecting those intersection calls with the original dup calls,
+    ## requiring full reciprocal coverage (-r 1) to find only those dup calls that are in the original set and NOT covered by a del call
+    input:
+        dup_merge = "../../scratch/interim/{sample}.dup.merged.bed",
+        del_merge = "../../scratch/interim/{sample}.del.merged.bed"
+    output:
+        dup_merge = "../../scratch/endpoints/{sample}.dup.merged.final.bed",
+        del_merge = "../../scratch/endpoints/{sample}.del.merged.final.bed"
+    shell:
+        "bedtools subtract -a {input.dup_merge} -b {input.del_merge}"
```

Comment:
Can you put comments at each step so these steps are split up a bit and easier to follow?